### ISSUE TITLE
Connection pool is not closed properly when deploying.

### ISSUE THREAD

**Issue by fl0x0** [steineey/node-red-contrib-mongodb4#18]
Hi,
I really like your mongodb node-red implementation. However when I use it and do multiple deployments, I reach my 500 connection max fairly quick. When I realaized that I switched to https://flows.nodered.org/node/node-red-contrib-mongodb-aleph without any issues for weeks. But the aleph implementation is a little short on the possible operations, so decided to give this one another shot as I had a fresh install of node-red as well. I immedatly see the connections piling up again with every deploy. Below is a screenshot from the atlas console after I triggered some deployments in node-red. No other client is connected to the cluster.

<img width="1088" alt="image" src="https://user-images.githubusercontent.com/35115362/230090259-0d5eb29f-bccf-47fe-8c96-3ccb75437cf6.png">

It would be awesome if you could look into this, I would really like to use your implementation going forward.

Similar to issue #14

**Comment by steineey** [comment]
Hi,
thank you for your feedback. This issue should be fixed since v1.8.1 but I will try to reproduce your problem and solve it. It would be great if you could name the versions of your installation: NodeJS, Node-RED and the version of this package you are using.
I do also use MongoDB Atlas but after redeploying my flows several times my connection count is still low. Are you sure that this issue is related to flow redeployment? How do you redeploy: Full or only modified?
The expected behaviour is that the node will close and reopen connections only on full deployment. The node is initializing a MongoClient instance per connection config. A default MongoClient instance can handle a connection pool of up to 100 connections. You can overwrite the connection options with this node for optimization, but I don't think that this solves your issue.

**Comment by fl0x0** [comment]
Hello and thank you for your quick response! 
I always used full deployment just because I left the setting as default. To your question if I am sure that this is related to deployment, I think there is a correlation with full deployments at least (see tests below). There seems to be a difference somewhre as I never had so many connections with the aleph mongodb implementation. Thank you very much for looking into this, I will do my best gathering insights that might help.

## Versions
Node-RED version: v3.0.2
Node.js  version: v16.16.0
node-red-contrib-mongodb4 version: 1.10.0


# Test
Just did a new test, see below.

## Steps I have taken to prep:
1. disable aleph mongodb config
2. reboot node-red to remove old hanging connections

## Test 1 with light deploy:
1. Trigger an update on the db with a mongodb4 node
2. Change a mongodb4 node position in node red
3. Deploy only changed node
4. Check connections metrics in Atlas
5. Repeat 1-4
==> Result: Only deplyoing the changed node does not affect the connections count

## Test 2 with full deploy:
1. Trigger an update on the db with a mongodb4 node
2. Change a mongodb4 node position in node red
3. **Full Deploy**
4. Check connections metrics in Atlas
5. Repeat 1-4
==> Result: Over the course of 10 min. the connection count grew significantly 

## Connections Graph
<img width="1096" alt="image" src="https://user-images.githubusercontent.com/35115362/230552234-b132587c-a54f-4ecb-b718-7a658839a3d5.png">

**Comment by steineey** [comment]
Hi,

I released v2.0.0 on npm. Please try this release. I hope this solves this issue.

## What's new?

* I upgrade to mongodb driver v5.2.0
* I added more connection options to have more control over the connection pool. Check out the configuration node. All options are explained in the readme and node documentation.
* I added a log entry to node-reds runtime log on client instance init and after client instance closed. 
* The node supports now `appName`. The node generates an automatic appName in combination with a connection pool session key. All connections to the db server are tagged with this app name. MongoDB 3.4 and newer will print this value in the server log upon establishing each connection. It is also recorded in the slow query log and profile collections. In your case you can check if there is more than one connection pool connected with your database and if old connections from old pools have terminated. The appName is printed to the nodered runtime log.

## Testing

I tested the connection pooling with a local single host Mongodb server and with an Atlas ReplicaSet. I did multiple full deployments and nodered runtime restarts. The connection count has never exploded like in your example.

On my own Mongodb server i can run to show all current connected connection pools:
```js
db.currentOp(true).inprog.reduce((accumulator, connection) => {
    const appName = connection.appName || "unknown";
    accumulator[appName] = (accumulator[appName] || 0) + 1;
    accumulator.totalCount ++;
    return accumulator;
  }, {totalCount: 0})
```
The node is closing all previous connections on full deployment or nodered runtime shutdown.

The difference between `node-red-contrib-mongodb-aleph` and my implementation is that aleph is creating a connection pool for each mongodb flow node. My implementation creates only one connection pool for each connection config node. I think that this is much more efficient. Because multiple flow nodes can use a single connection pool and also reuse connections.

Please let me know if the latest release solves your issue.

**Comment by fl0x0** [comment]
You are awesome! Just redid my test scenario and the connection count is steady now. Thank you so much!
